components/AccountingCategorySelect.tsx

```diff
@@ -423,79 +418,80 @@ const AccountingCategorySelect = ({
               )}
               <CommandGroup
                 heading={
-                  !suggestedOptions ? null : intl.formatMessage({ defaultMessage: 'All Categories', id: '1X6HtI' })
+                  !useSeparatePredictionsCommandGroup
+                    ? undefined
+                    : intl.formatMessage({ defaultMessage: 'All Categories', id: '1X6HtI' })
                 }
               >
-                {Object.entries(options).map(([categoryId, { label }]) => {
-                  const isSelected = selectedCategory?.id === categoryId;
-                  const isPrediction = predictions?.some(prediction => prediction.id === categoryId);
+                {options.map(({ label, key, value }) => {
+                  const isSelected = selectedCategory?.id === key;
+                  const isPrediction = suggestedOptions.some(option => option.key === key);
                   return (
-                    <React.Fragment key={categoryId}>
-                      <CommandItem
-                        value={categoryId}
-                        onSelect={categoryId => {
-                          triggerChange(options[categoryId].value);
-                          setOpen(false);
-                        }}
-                      >
-                        <div className="flex flex-1 items-center justify-between">
-                          <span
-                            className={
-                              // If there are predictions, grey out the categories that are not selected or predicted
-                              isSelected || isPrediction || !hasPredictions || predictionStyle === 'inline-preload'
-                                ? 'text-foreground'
-                                : 'text-muted-foreground'
-                            }
-                          >
-                            {label}
-                          </span>
-                          <div className="flex items-center gap-1 pt-0.5">
-                            {isSelected && <Check size={16} className="ml-2 inline-block" />}
-
-                            {isPrediction && (
-                              <span
-                                className="text-right text-xs text-gray-500"
-                                title={intl.formatMessage({ defaultMessage: 'Suggested', id: 'a0lFbM' })}
-                              >
-                                <Sparkles size={16} className="mr-1 inline-block text-yellow-500" strokeWidth={1.5} />
-                              </span>
-                            )}
-                          </div>
+                    <CommandItem
+                      key={key || 'none'} // `CommandItem` doesn't like nil key/value
+                      value={key || 'none'}
```

Comment:
This `none` is the actual fix for https://opencollective.slack.com/archives/GFH4N961L/p1721668122051539